parser/src/parser.cpp

```diff
@@ -204,16 +213,32 @@ int main(int argc, char* argv[])
     return 1;
   }
 
-  //--- Check workspace and project directory ---//
+  //--- Check database and project directory existence ---//
+
+  bool isNewDb = cc::util::connectDatabase(
+    vm["database"].as<std::string>(), false) ? false : true;
+  bool isNewProject = !checkProjectDir(vm);
+
+  if (isNewProject && !isNewDb || !isNewProject && isNewDb)
```

Comment:
`&&` within `||` will trigger compiler warnings, please put the subformulae of `||` in parentheses.

Reply:
Also, I think this condition is the same as `isNewProject XOR isNewDb`.